### PR TITLE
MWPW-136270 Homepage blocks

### PR DESCRIPTION
**Pull request by vgoodric**
Resolves: [MWPW-136270](https://jira.corp.adobe.com/browse/MWPW-136270)

- Adds 2 new repository-specific blocks: homepage-brick and homepage-link-bar
- Adds CSS support for masonry style in section-metadata

Test URLs:
Before:
[https://main--homepage--adobecom.hlx.page/homepage/index-loggedout](https://main--homepage--adobecom.hlx.page/homepage/index-loggedout)

After:
[https://homepage-brick--homepage--adobecom.hlx.page/homepage/index-loggedout](https://homepage-brick--homepage--adobecom.hlx.page/homepage/index-loggedout)

### CODE REVIEW

File: test/blocks/homepage-brick/homepage-brick.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      expect(directChildren.length).to.equal(3);
+      expect(directChildren[0].className).to.equal('background first-background');
+      expect(directChildren[1].className).to.equal('background');
+      expect(directChildren[2].className).to.equal('foreground');
+      const h3 = block.querySelector('h3');
+      expect(h3).to.be.exist;
+      const link = block.querySelector('div.click-link');
+      expect(link).to.be.exist;
+      const { x, y } = getMiddleOfElement(block);
+      await sendMouse({ type: 'click', position: [x, y] });
```
Comment (seanchoi-dev) [inline test/blocks/homepage-brick/homepage-brick.test.js:37]: I know you will revisit this again, but wondering what will be the expect result of click event.

File: homepage/blocks/homepage-brick/homepage-brick.js [hunk 2]
```diff
@@ -159,6 +115,21 @@ export default async function init(el) {
     const [highlight, ...tail] = rows;
     highlight.classList.add('highlight-row');
     rows = tail;
+  } else {
+    let [head, ...tail] = rows;
+    const classList = el.className.split(' ');
+    classList.splice(1, 0, 'click');
+    el.className = classList.join(' ');
```
Comment (seanchoi-dev) [inline homepage/blocks/homepage-brick/homepage-brick.js:122]: Is there a reason why we want to add `click` as second classname instead simply add 'click' for the element?
like:
```suggestion
    el.classList.add('click');
```

File: homepage/blocks/homepage-link-bar/homepage-link-bar.css [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+  .homepage-link-bar > div.header {
+    white-space: nowrap;
+  }
+
+  .homepage-link-bar.button-count-6 > div.header {
+    flex-basis: 100%;
+  }
+}
+
+@media screen and (min-width: 1300px) {
```
Comment (seanchoi-dev) [inline homepage/blocks/homepage-link-bar/homepage-link-bar.css:113]: I see other sites use 1200px for desktop, do we have standard breakpoints in our sites? Maybe consonant?

**Review by seanchoi-dev — COMMENTED** [review]
Thanks for doing this. 
I added few comments (mostly questions)

File: test/blocks/homepage-brick/homepage-brick.test.js (hunk 1 above)
Comment (vgoodric) [reply test/blocks/homepage-brick/homepage-brick.test.js:37]: clicking anywhere in the pod should take the user to the destination of the link.  So the href of the link is moved to a data attribute on the foreground element, and a click event makes the element behave like a link.  I had trouble getting the analytics and cosmetic of the block to look the way I wanted if I actually made the foreground a link.  But maybe we can revisit that.

File: homepage/blocks/homepage-brick/homepage-brick.js (hunk 2 above)
Comment (vgoodric) [reply homepage/blocks/homepage-brick/homepage-brick.js:122]: there WAS a good reason.  But we updated the way the new analytics taxonomy will work earlier this week, and it makes this code unnecessary.

File: homepage/blocks/homepage-link-bar/homepage-link-bar.css (hunk 3 above)
Comment (vgoodric) [reply homepage/blocks/homepage-link-bar/homepage-link-bar.css:113]: Well, the 1300px breakpoint was only for the fringe case of 6 buttons in the block.  It needed to wrap a little sooner to still look good.  The real issue was the media query above it for 1025px.  That's when "desktop" was starting.  But you're right so  I moved that breakpoint to 1200.  But I'd like to keep the 1300px additional breakpoint for 6 buttons.

Comment (vgoodric) [reply homepage/blocks/homepage-link-bar/homepage-link-bar.css:114]: Here's a preview page a made a while back to see the behavior of the block with different numbers of buttons: https://homepage-brick--homepage--adobecom.hlx.page/homepage/drafts/vgoodrich/link-bars

**Review by Blainegunn — APPROVED** [review]

**Review by seanchoi-dev — APPROVED** [review]

**Pull request merged**